test/controllers/competencies_controller_test.rb

```diff
@@ -0,0 +1,78 @@
+require 'test_helper'
+
+class CompetenciesControllerTest < ActionController::TestCase
+  # Test all competency controller actions and methods
+  setup do
+    create_competencies
+  end
+
+  # Test that the index page displays with proper variables assigned
+  test "should get index" do
+    get :index
+    assert_response :success
+    assert_not_nil assigns(:active_competencies)
+    assert_not_nil assigns(:inactive_competencies)
+  end
+
+  # Test that the show page displays with proper object assigned
+  test "should show competency" do
+    get :show, id: @communication
+    assert_not_nil assigns(:competency)
+    assert_response :success
+  end
+
+  # Test that the new page displays
+  test "should get new" do
+    get :new
+    assert_response :success
+  end
+
+  # Test that the create action actually creates a valid object
+  test "should create competency" do
+    assert_difference('Competency.count', 1) do
+      post :create, competency: { name: "Test Competency", description: "Test Competency Description" }
+    end
+
+    assert_redirected_to competency_path(assigns(:competency))
+  end
+
+  # Test that the create action doesn't create an invalid object
+  test "should not create invalid competency" do
+    assert_no_difference('Competency.count') do
+      post :create, competency: { name: "", description: "Bad Test Competency Description" }
+    end
+
+    assert_response :success

```

Comment:
yea response should still be success, because the controller (or response) doesn't fail, it just redirects to the new form again if the model isn't valid